caddyconfig/httploader.go

```diff
@@ -116,10 +130,22 @@ func (hl HTTPLoader) LoadConfig(ctx caddy.Context) ([]byte, error) {
 		ctx.Logger(hl).Warn(warn.String())
 	}
 
+	// Upon successful response, save the new time of last modification, if any.
+	lastModified := resp.Header.Get("Last-Modified")
+	if lastModified != "" {
+		t, err := time.Parse(timeLayout, lastModified)
+		if err != nil {
+			ctx.Logger(hl).Debug("failed to parse Last-Modified header", zap.Error(err))
+		}
+		if !t.IsZero() {
+			hl.lastModified = t
```

Comment:
I was just wondering, does storing the lastModified field in h1 serve any purpose?
Caddy creates a new configloader instance on every change detected. So the value we store here will be lost as soon as we return a new result.

Reply:
> Caddy creates a new configloader instance on every change detected.

Yeah, that is the case.

The `lastModified` field — initially being zero — will typically be saved for the first polling (just `load_delay` time later after the associated loader instance was created). Then in the subsequent queries, the `If-Modified-Since` header can be filled in correctly.

Of course, when the loader finally detects a config change (i.e. receives a 200 response for the second time), the last save will be lost as you described.